### PR TITLE
Dgi configuration setup

### PR DESCRIPTION
**Pull request by bradbergeron-us**
## Description of change
Integration of a new service DGIB. This is an initial PR to setup the base settings and configuration of being able to request data from this service. When the controller methods are implemented to render the responses from the service objects they will be sitting behind a flipper feature flag that is set to off in production environment. Goal of this PR is to be able to test out a couple requests/routes on staging with our initial configuration.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -995,6 +995,15 @@ genisis:
   user: TestUser
   pass: bogus
 
+# Settings for connecting AFS Veteran Services
+dgi:
```
Comment (LindseySaari) [inline config/settings.yml:999]: Did you add these setting to the various dev, staging, prod, sandbox files in the devops repo? I didn't see any links in the PR description

File: modules/meb_api/lib/authentication_token_service.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module MebApi
+  class AuthenticationTokenService
+    ALGORITHM_TYPE = 'RS256'
```
Comment (LindseySaari) [inline modules/meb_api/lib/authentication_token_service.rb:5]: Do you want to namespace the services, config, etc classes with the v0 namespace?

File: modules/meb_api/app/controllers/meb_api/v0/education_benefits_controller.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require 'dgi/eligibility/service'
```
Comment (LindseySaari) [inline modules/meb_api/app/controllers/meb_api/v0/education_benefits_controller.rb:3]: Does the meb_api module have API docs as of yet? I'm not seeing any currently

File: config/settings.yml (hunk 1 above)
Comment (bradbergeron-us) [reply config/settings.yml:999]: This is the PR that added those to staging, and sandbox. [DevOps PR for staging and sandbox env](https://github.com/department-of-veterans-affairs/devops/pull/10152). There is a pending PR for Prod [DevOps PR for Prod is pending](https://github.com/department-of-veterans-affairs/devops/pull/10153)

File: modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb [hunk 4]
```diff
@@ -20,10 +31,15 @@
...
-      get '/meb_api/v0/eligibility'
-      expect(response).to have_http_status(:ok)
-      JSON.parse(response.body).to_yaml
+    context 'Veteran who has benefit eligibility' do
+      it 'returns a 200 with eligibility data' do
+        VCR.use_cassette('dgi/get_eligibility') do
+          get '/meb_api/v0/eligibility'
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to be_a(String)
+          # expect(response).to match_response_schema('dgi/eligibility_response', { strict: false })
```
Comment (LindseySaari) [inline modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: Is line 40 meant to be commented out here?

File: modules/meb_api/app/controllers/meb_api/v0/education_benefits_controller.rb (hunk 3 above)
Comment (bradbergeron-us) [reply modules/meb_api/app/controllers/meb_api/v0/education_benefits_controller.rb:3]: No we don't have API/Swagger docs as of yet but are aware they will be needed as the API gets finalized. These routes are behind a feature flag that is disabled in production. The goal for this is to make sure that the two environments can communicate in sandbox/staging environment.

File: modules/meb_api/lib/dgi/automation/configuration.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        def base_path
+          Settings.dgi['veteran-services'].url.to_s
+        end
+
+        def service_name
+          'DGI/Automation'
+        end
+
+        def mock_enabled?
+          Settings.dgi['veteran-services'].mock || false
```
Comment (LindseySaari) [inline modules/meb_api/lib/dgi/automation/configuration.rb:18]: Can this be Settings.dgi.veteran-services.mock ?

File: modules/meb_api/lib/dgi/automation/configuration.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'dgi/configuration'
+
+module MebApi
+  module DGI
+    module Automation
+      class Configuration < MebApi::DGI::Configuration
+        def base_path
+          Settings.dgi['veteran-services'].url.to_s
```
Comment (LindseySaari) [inline modules/meb_api/lib/dgi/automation/configuration.rb:10]: Can this be Settings.dgi.veteran-services.url ?

File: modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb (hunk 4 above)
Comment (bradbergeron-us) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: It was intentionally commented out as it was causing the test suite to fail. I'm open to feedback on getting the match_response_schema method working.

File: modules/meb_api/lib/dgi/eligibility/configuration.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'dgi/configuration'
+
+module MebApi
+  module DGI
+    module Eligibility
+      class Configuration < MebApi::DGI::Configuration
+        def base_path
+          Settings.dgi['veteran-services'].url.to_s
```
Comment (LindseySaari) [inline modules/meb_api/lib/dgi/eligibility/configuration.rb:10]: Same comments here as the modules/meb_api/lib/dgi/automation/configuration.rb file.

File: config/settings.yml (hunk 1 above)
Comment (LindseySaari) [reply config/settings.yml:999]: Thanks!

File: modules/meb_api/lib/dgi/automation/configuration.rb (hunk 6 above)
Comment (LindseySaari) [reply modules/meb_api/lib/dgi/automation/configuration.rb:10]: Ah ok, it was a rubocop correction. Totally fine!

File: modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb (hunk 4 above)
Comment (LindseySaari) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: What were the errors?

Comment (bradbergeron-us) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: `     Failure/Error: JSON::Validator.validate!(schema_path.to_s, json, { strict: true }.merge(opts))

     JSON::Schema::JsonParseError:
       expected hash key, not a hash close (after properties.data.properties.type.type.properties.chapter) at line 32, column 16 [par
     # ./spec/support/matchers/schema_matchers.rb:10:in `valid_against_schema?'`

Comment (LindseySaari) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: Ok, I feel like the schema match is still a valid test to have passing though.

Comment (bradbergeron-us) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: Agreed. Does that error give you any idea what the root cause may be?

Comment (LindseySaari) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb:40]: Does [this](https://stackoverflow.com/questions/7543779/ruby-json-parse-changes-hash-keys) maybe help?

Comment (bradbergeron-us) [reply modules/meb_api/spec/requests/meb_api/v0/education_benefits_spec.rb]: I traced it back to the schema file itself. Should be addressed in the latest commit on this branch.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**